enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,221 @@
+---
+title: cli-manager
+authors:
+  - "@sallyom"
+  - "@deejross"
+reviewers:
+  - "@soltysh"
+approvers:
+  - "@soltysh"
+creation-date: 2021-10-06
+last-updated: 2021-10-06
+status: implementable
+---
+
+# OpenShift CLI Manager
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal is describing the mechanism for how authors of a Command Line Interface (CLI) such as oc, kubectl, odo, istio, tekton, or knative,
+can deliver tools to OpenShift clusters.  A feature is needed to manage various CLIs available for OpenShift and related services.  The goal is for
+ a connected user to discover, install, and upgrade tools that are compatible with the current cluster version easily and from a single location.
+
+Each component is responsible for building and publishing its artifacts and registering information regarding supplied binaries.
+Currently, that location is [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/),
+and this makes it difficult for disconnected installations to mirror them.  
+
+`oc` will retrieve binaries from images that package a CLI's artifacts. Currently, we provide disconnected environments 
+cli-artifacts and oc download links that do not require anything outside the cluster.  The goal is to provide the same for other
+binaries/tools/CLIs. Also, these artifacts will be accessible via `oc` as well as through the console via a fileserver.
+
+`oc` will provide the mechanism to provide, list, install, and upgrade OpenShift supported CLIs compatible with   
+a connected cluster. Through a ClusterCLI [Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/),
+each CLI will provide necessary information about its provided artifacts, including its description and location within the image.  The ClusterCLI
+for oc, helm, and odo custom resources as well as a CRD controller to manage these will serve as a reference consumer implementation for other CLIs to 
+follow. Currently. the `cli-artifacts` image stream in the `openshift namespace` is shipped with the release payload. This is where the artifacts
+for `oc` will be extracted from. If other CLIs choose to offer artifacts in this way, they can opt in by supplying an artifacts image to be
+shipped with the release payload.
```

Comment:
Placing them in the release payload ties them to the release schedule. Is that what we want?

We're actively trying to limit the growth of the release payload, so if there are ways to achieve the goals of this enhancement that avoids adding anything we should consider those approaches.